hellabot.go

```diff
@@ -236,116 +189,123 @@ func (irc *IrcCon) WaitFor(filter func(*Message) bool) {
 }
 
 // A basic set of registration commands
-func (irc *IrcCon) StandardRegistration() {
+func (bot *Bot) StandardRegistration() {
 	//Server registration
-	if irc.Password != "" {
-		irc.Send("PASS " + irc.Password)
+	if bot.Password != "" {
+		bot.Send("PASS " + bot.Password)
 	}
-	irc.sendUserCommand(irc.nick, irc.nick, "8")
-	irc.SetNick(irc.nick)
+	bot.Debug("Sending standard registration")
+	bot.sendUserCommand(bot.Nick, bot.Nick, "8")
+	bot.SetNick(bot.Nick)
 }
 
 // Set username, real name, and mode
-func (irc *IrcCon) sendUserCommand(user, realname, mode string) {
-	irc.Send(fmt.Sprintf("USER %s %s * :%s", user, mode, realname))
+func (bot *Bot) sendUserCommand(user, realname, mode string) {
+	bot.Send(fmt.Sprintf("USER %s %s * :%s", user, mode, realname))
 }
 
-func (irc *IrcCon) SetNick(nick string) {
-	irc.Send(fmt.Sprintf("NICK %s", nick))
+func (bot *Bot) SetNick(nick string) {
+	bot.Nick = nick
+	bot.Send(fmt.Sprintf("NICK %s", nick))
 }
 
 // Start up servers various running methods
-func (irc *IrcCon) Start() {
-	irc.Log(LTrace, "Start bot processes.")
+func (bot *Bot) Start() {
+	bot.Debug("Start bot processes.")
 
-	go irc.handleIncomingMessages()
-	go irc.handleOutgoingMessages()
+	go bot.handleIncomingMessages()
+	go bot.handleOutgoingMessages()
 
-	go irc.StartUnixListener()
+	go bot.StartUnixListener()
 
 	// Only register on an initial connection
-	if !irc.reconnect {
-		if irc.DoSasl {
-			irc.SASLAuthenticate(irc.nick, irc.Password)
+	if !bot.reconnecting {
+		if bot.SASL {
+			bot.SASLAuthenticate(bot.Nick, bot.Password)
 		} else {
-			irc.StandardRegistration()
+			bot.StandardRegistration()
 		}
 	}
-
-	for _, s := range irc.JoinAfterConnection {
-		irc.Join(s)
-	}
 }
 
 // Send a message to 'who' (user or channel)
-func (irc *IrcCon) Msg(who, text string) {
+func (bot *Bot) Msg(who, text string) {
 	// if len(text) == 0, return instead of trying to send a empty message
 	if len(text) == 0 {
 		return
 	}
 	for len(text) > 400 {
-		irc.Send("PRIVMSG " + who + " :" + text[:400])
+		bot.Send("PRIVMSG " + who + " :" + text[:400])
 		text = text[400:]
 	}
-	irc.Send("PRIVMSG " + who + " :" + text)
+	bot.Send("PRIVMSG " + who + " :" + text)
 }
 
 // Notice sends a NOTICE message to 'who' (user or channel)
-func (irc *IrcCon) Notice(who, text string) {
+func (bot *Bot) Notice(who, text string) {
+	// if len(text) == 0, return instead of trying to send a empty notice
+	if len(text) == 0 {
+		return
```

Comment:
I have removed those checks now. I added them because at some point I was lazy and I kept getting IRC errors about empty messages.